framerail/src/lib/server/page/edit.ts

```diff
@@ -0,0 +1,42 @@
+import { wellfetch } from "$lib/server/deepwell/index.ts"
+import type { Optional } from "$lib/types.ts"
+
+export async function pageEdit(
+  siteId: number,
+  pageId: Optional<number>,
+  slug: string,
+  revisionComments: Optional<string>,
+  wikitext: string,
+  title: string,
+  altTitle: string,
+  tags: string[]
+): Promise<object> {
+  let endpoint = "/page"
+  if (!pageId) {
+    // Assume page creation
+    endpoint += "/create"
+  }
+  const response = await wellfetch(endpoint, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify({
+      siteId,
+      page: pageId ?? slug,
+      slug,
```

Comment:
Does create require page ID at all?

Reply:
This method is combined for page creation and edit, so I made it to use pageId first for page edit, and fall back to slug for page creation.